src/overlays/actors/ovl_Obj_Bean/z_obj_bean.c

```diff
@@ -27,160 +72,895 @@ const ActorInit Obj_Bean_InitVars = {
+        BgCheck_EntityRaycastFloor5(&globalCtx->colCtx, &this->dyna.actor.floorPoly, &sp20, &this->dyna.actor, &sp24);
+}
+
+s32 func_80936D58(ObjBean* this, GlobalContext* globalCtx) {
+    static Vec3f D_80939018 = { 0.0f, 30.0f, 0.0f };
+    s32 pad;
+    s32 spB8;
+    Vec3f spAC;
+    Vec3f spA0;
+    Vec3f sp94;
+    Vec3f sp88;
+    MtxF sp48;
+
+    Matrix_RotateY(this->dyna.actor.shape.rot.y, MTXMODE_NEW);
+    Matrix_InsertXRotation_s(this->dyna.actor.shape.rot.x, MTXMODE_APPLY);
+    Matrix_InsertZRotation_s(this->dyna.actor.shape.rot.z, MTXMODE_APPLY);
+    Matrix_MultiplyVector3fByState(&D_80939018, &spAC);
+    Math_Vec3f_Sum(&this->dyna.actor.world.pos, &spAC, &spA0);
+    Math_Vec3f_Diff(&this->dyna.actor.world.pos, &spAC, &sp94);
+
+    if (BgCheck_EntityLineTest2(&globalCtx->colCtx, &spA0, &sp94, &sp88, &this->dyna.actor.floorPoly, 1, 1, 1, 1, &spB8,
+                                &this->dyna.actor)) {
+        this->dyna.actor.world.pos.x = (COLPOLY_GET_NORMAL(this->dyna.actor.floorPoly->normal.x) * 1.9f) + sp88.x;
+        this->dyna.actor.world.pos.y = (COLPOLY_GET_NORMAL(this->dyna.actor.floorPoly->normal.y) * 1.9f) + sp88.y;
+        this->dyna.actor.world.pos.z = (COLPOLY_GET_NORMAL(this->dyna.actor.floorPoly->normal.z) * 1.9f) + sp88.z;
+        func_800C0094(this->dyna.actor.floorPoly, this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y,
+                      this->dyna.actor.world.pos.z, &sp48);
+        Matrix_SetCurrentState(&sp48);
+        Matrix_RotateY(this->dyna.actor.shape.rot.y, MTXMODE_APPLY);
+        Matrix_CopyCurrentState(&sp48);
+        func_8018219C(&sp48, &this->dyna.actor.shape.rot, 0);
+        return true;
+    }
+    return false;
+}
+
+void func_80936F04(ObjBean* this) {
+    this->unk_1AC = this->unk_1AE = this->unk_1B0 = 0;
+    this->unk_1C8 = 0.0f;
+}
+
+void func_80936F24(ObjBean* this) {
+    this->unk_1AC += 0xB6;
+    this->unk_1AE += 0xFB;
+    this->unk_1B0 += 0x64;
+    Math_StepToF(&this->unk_1C8, 2.0f, 0.1f);
+    this->unk_1CC = (Math_SinS(this->unk_1AC * 3) + Math_SinS(this->unk_1AE * 3)) * this->unk_1C8;
+    this->unk_1D0 = (Math_CosS(this->unk_1AC * 4) + Math_CosS(this->unk_1AE * 4)) * this->unk_1C8;
+    this->dyna.actor.scale.z =
+        ((Math_SinS(this->unk_1B0 * 5) * 0.06f) + (Math_SinS(this->unk_1AE * 8) * 0.01f) + 1.07f) * 0.1f;
+    this->dyna.actor.scale.x = this->dyna.actor.scale.z;
+    this->dyna.actor.scale.y = ((Math_CosS(this->unk_1B0 * 10) * 0.2f) + 1.0f) * 0.1f;
+    this->dyna.actor.shape.rot.y = (this->dyna.actor.home.rot.y + (s32)(Math_SinS(this->unk_1AC * 3) * 1000.0f)) +
+                                   (s32)(Math_SinS(this->unk_1B0 * 2) * 2100.0f);
+}
+
+void func_80937130(ObjBean* this) {
+    this->unk_1AC = this->unk_1AE = this->unk_1B0 = 0;
+    Actor_SetScale(&this->dyna.actor, 0.0f);
+}
+
+void func_80937160(ObjBean* this) {
+    this->unk_1AC += 0x384;
+    if (this->unk_1AC >= 0x6000) {
+        this->unk_1AC = 0x5FFF;
+    }
```

Comment:
```suggestion
    if (this->unk_1AC >= 0x6000) {
        this->unk_1AC = 0x5FFF;
    }
```
Does a `CLAMP_MAX(this->unk_1AC, 0x5FFF);` work by chance?

Reply:
It doesn't, it keeps the unconditional branch.